src/core/instances.jl

```diff
@@ -168,10 +168,33 @@ function Base.getindex(obj::AbstractCompositeComponentInstance, comp_name::Symbo
     return compinstance(obj, comp_name)
 end
 
-function _get_datum(ci::AbstractComponentInstance, datum_name::Symbol)
+# LFR TODO some of this is very similar to _get_datum for leaf component instance, 
+# but in some ways it seems clearner to separate them
+function _get_datum(ci::CompositeComponentInstance, datum_name::Symbol)
     vars = variables(ci)
 
-    if datum_name in names(vars)
+    if datum_name in keys(vars) # could merge with below if made names(NamedTuple) = keys(NamedTuple)
+        which = vars
+    else
+        pars = parameters(ci)
+        if datum_name in keys(pars) # could merge with below if made names(NamedTuple) = keys(NamedTuple)
+            which = pars
+        else
+            error("$datum_name is not a parameter or a variable in component $(ci.comp_path).")
+        end
+    end
+    
+    ref = getproperty(which, datum_name)
+    
+    # LFR TODO get the comp_name and datum_name to use ...
+    return _get_datum(ci.comps_dict[comp_name, datum_name])
```

Comment:
I hoped this could just be `_get_datum(ci.comps_dict[ref.comp_name], ref.datum_name)` but see comment above